lib/codecs/tests/avro.rs

```diff
@@ -0,0 +1,75 @@
+use bytes::{Bytes, BytesMut};
+use codecs::{
+    decoding::format::AvroDeserializerConfig, decoding::format::Deserializer,
+    encoding::format::AvroSerializerConfig,
+};
+use rstest::*;
+use similar_asserts::assert_eq;
+use std::{
+    fs::File,
+    io::Read,
+    path::{Path, PathBuf},
+    str::from_utf8,
+};
+use tokio_util::codec::Encoder;
+use vector_core::{config::LogNamespace, event::Event};
+
+#[rstest]
+#[case(true)]
+#[case(false)]
+fn roundtrip_avro_fixtures(
+    #[files("tests/data/avro/generated/*.avro")]
+    #[exclude(".*(date|decimal_var|fixed|time_millis|timestamp_micros|timestamp_millis).avro")]
```

Comment:
In the implementation we return an error for:    

* `Decimal`
* `Duration`
* `LocalTimestampMillis`
* `LocalTimestampMicros`

In this exclusion list I see `date`, `decimal_var` and `fixed`. Are these failing or fall under unsupported? 

For example, we do support `AvroValue::Fixed` but if the test fails, it means there is a bug.

Reply:
So this seems to be an issue converting from Vrl to Avro

So date is not returned as a Date, but as a Long. Same issue for time_millis.

timestamp_micros and timestamp_millis are longs in Avro but maped to timestamps for Vrl, so when trying to change them back to Avro they fail with unexpected type.

Not quite sure how to fix it other than finding a way to know what type to convert to in encode in `lib/codecs/src/encoding/format/avro.rs` - maybe we can set the schema for the `apache_avro::to_value` somehow so it knows to change timestamp to long agin.

I can set them as unsupported for now if you want to, then i can open a new pr if i find a way to fix this.

Reply:
Let's mark them as unsupported to unblock this PR. Please include the info you shared here in the docs.

Reply:
Returning error on them now, have added them to the unsupported list and shortly explained why with a link to my comment above.